pkg/controller/datavolume-controller.go

```diff
@@ -383,6 +460,8 @@ func (c *DataVolumeController) updateDataVolumeStatus(dataVolume *cdiv1.DataVolu
 	// Only update the object if something actually changed in the status.
 	if !reflect.DeepEqual(dataVolume.Status, dataVolumeCopy.Status) {
 		_, err = c.cdiClientSet.CdiV1alpha1().DataVolumes(dataVolume.Namespace).Update(dataVolumeCopy)
+		// Emit the event only when the status change happens, not every time
+		c.recorder.Event(dataVolume, eventType, reason, message)
```

Comment:
OH, I see; so you're capturing those responses to feed to the recorder here.  I'm wondering if there might be a good reason to make this even a struct?  So the updatexxxPhase could return an event type that you could share and then pass in to the recorder?